server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTree.java

```diff
@@ -367,7 +370,15 @@ public IMeasurementMNode createTimeseries(
       throw new PathAlreadyExistException(cur.getFullPath());
     }
 
-    if (upperTemplate != null && !upperTemplate.isCompatible(path)) {
+    if (upperTemplate != null
+        && upperTemplate.isDirectNodeInTemplate(nodeNames[nodeNames.length - 1])) {
+      throw new PathAlreadyExistException(
+          path.getFullPath() + " ( which is incompatible with template )");
+    }
+
+    // Quick patch for tree structure template
```

Comment:
fast return?

Reply:
It's not a fast return, but should get clear with process of createTimeseries:

If I have template [vehicle.GPS.x, vehicle.GPS.y] mounted to " root.d ",
Can I createTimeseries root.d.vehicle.speed?

If not, current code is to check the direct node

Reply:
ok, we could reject currently